uaclient/jobs/update_messaging.py

```diff
@@ -417,3 +424,32 @@ def refresh_motd():
             logging.exception(exc)
 
     system.subp(["sudo", "systemctl", "restart", "motd-news.service"])
+
+
+def update_contract_expiry(cfg: config.UAConfig):
+    orig_token = cfg.machine_token
+    machine_token = orig_token.get("machineToken", "")
+    contract_id = (
+        orig_token.get("machineTokenInfo", {})
+        .get("contractInfo", {})
+        .get("id", "")
+    )
+    contract_client = contract.UAContractClient(cfg)
+    resp = contract_client.get_updated_contract_info(
```

Comment:
If we can't get the `contract id` here, I don't think we should make the request, because it will only fail here and raise an Exception to the job. I think we should only proceed if the id is valid

Reply:
Makes sense. I think we should also update `is_contract_changed` as well.